app/bot/openai.go

```diff
@@ -92,6 +90,64 @@ func (o *OpenAI) request(text string) (react bool, reqText string) {
 	return false, ""
 }
 
+func (o *OpenAI) checkRequest(msg Message, text string) (ok bool, response Response) {
```

Comment:
to me this function doesn't feel right because it returns the full Response. This is a little bit too much responsibility for smth what suppose to check things. I'd rather would have it to return (ok bool, banText string) and handle all the response part on the caller side

Reply:
up to you, i don't have strong preference.